### PR TITLE
solution

### PR DESCRIPTION
**Pull request by GoldenSpade**

### CODE REVIEW

**Review by stas-dukach — APPROVED** [review]
good

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -65,7 +65,45 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
-}
+  const states = [];
+  let stateClone = {
+    ...state,
+  };
```
Comment (varseniuk) [inline src/transformStateWithClones.js:71]:
```suggestion
  let stateClone = { ...state };
```
You can write it in one line but it's ok

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -65,7 +65,45 @@
...
+        break;
+
+      case 'clear':
+        clear();
+        break;
+    };
+    states.push({ ...stateClone });
+  };
+
+  function add(properties) {
```
Comment (varseniuk) [inline src/transformStateWithClones.js:90]:
```suggestion
  function add(properties) {
```
👍

File: src/transformStateWithClones.js [hunk 3]
```diff
@@ -65,7 +65,45 @@
...
+        break;
+
+      case 'removeProperties':
+        remove(transform.properties);
+        break;
+
+      case 'clear':
+        clear();
+        break;
+    };
```
Comment (varseniuk) [inline src/transformStateWithClones.js:86]: add `default` case

**Review by varseniuk — APPROVED** [review]
Good! 👍 
But don't forget to add `default` case to your `switch` statement

**Review by mykola-yarchuk — CHANGES_REQUESTED** [review]
test

**Review by mykola-yarchuk — APPROVED** [review]

**Pull request closed without merging**